src/serializer/ResidualHeapVisitor.js

```diff
@@ -928,9 +930,17 @@ export class ResidualHeapVisitor {
       oldCreatedObjects = this.createdObjects;
       this.createdObjects = new Set([...oldCreatedObjects, ...generator.effectsToApply[4]]);
     }
+    let lastGenerator = this.parentGenerator;
+    if (lastGenerator !== undefined && lastGenerator._name !== "main") {
+      this.generatorParents.set(generator, lastGenerator);
```

Comment:
`createGeneratorVisitCallbacks.visitGenerator` already doing for all nested generators --- nested within the main code, or additional functions.

I guess the intention of your change is to link up nested additional functions, but also linking additional functions to the main code? This makes strong assumptions on how `visitGenerator` is called --- I am not sure if they hold. Also, while this might make `commonAncestor` probing code succeed, it's also likely masking mismatches in expectations in `isReferencedOnlyByAdditionalFunction` --- this function is used in a few places, and it's all about having a *single* additional function. If a value is held on to by multiple (nested) additional functions, all those design assumptions get out of whack...